flake.nix

```diff
@@ -0,0 +1,57 @@
+{
+  description = "Zoomer application for Linux";
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = {
+    self,
+    nixpkgs,
+    flake-utils,
+  }:
+    flake-utils.lib.eachDefaultSystem (
+      system: let
+        pkgs = nixpkgs.legacyPackages.${system};
+      in {
+        formatter = pkgs.alejandra;
+        packages = rec {
+          boomer = let
+            opengl = pkgs.nimPackages.buildNimPackage rec {
+              pname = "opengl";
+              version = "1.2.8";
+              src = pkgs.fetchFromGitHub {
+                owner = "nim-lang";
+                repo = "opengl";
+                rev = version;
+                sha256 = "sha256-OHdoPJsmCFdyKV7FGd/r+6nh6NRF7TPhuAx7o/VpiDg=";
+              };
+            };
+          in
+            pkgs.nimPackages.buildNimPackage {
+              pname = "boomer";
+              version = "0.0.1";
+              nimBinOnly = true;
+
+              src = ./.;
+
+              buildInputs = with pkgs.nimPackages; [
+                x11
+                opengl
+              ];
+              propagatedBuildInputs = with pkgs; [
+                xorg.libX11
+                xorg.libXrandr
+                xorg.libXext
+                libGL
+                libGLU
+              ];
+            };
```

Comment:
No `meta.licence` or `meta.mainProgram`?

Reply:
> `meta.mainProgram` is not specified when the binary name matches the derivation name

That was because tools would assume that the main program is the same as the pname. Now, that is only the case for `nix run`, `lib.getExe` used to make that assumption but it's a bad assumption and was deprecated. https://github.com/NixOS/nixpkgs/pull/246386

Reply:
Thanks for the reference! I've added `mainProgram`